app/src/main/java/com/strv/movies/ui/login/MoviesLoginScreen.kt

```diff
@@ -0,0 +1,188 @@
+import androidx.compose.runtime.*
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.strv.movies.R
+import com.strv.movies.ui.error.ErrorScreen
+import com.strv.movies.ui.loading.LoadingScreen
+
+@Composable
+fun MoviesLoginScreen(
+    navigateToMovieList: () -> Unit,
+    viewModel: MoviesLoginViewModel = viewModel()
+) {
+    val viewState by viewModel.viewState.collectAsState()
+
+    if (viewState.loading) {
+        LoadingScreen()
+    } else if (viewState.error != null) {
+        ErrorScreen(errorMessage = viewState.error!!)
+    } else {
+        LoginElements(navigateToMovieList)
+
+    }
+}
+
+@Composable
+private fun LoginElements(navigateToMovieList: () -> Unit) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier
+            .padding(start = 40.dp, bottom = 20.dp, top = 20.dp, end = 40.dp)
+            .fillMaxWidth()
+    ) {
+        EmailField()
+        PasswordField()
+        LoginInWithSmileButton(navigateToMovieList)
+        ContinueWithFacebookButton(navigateToMovieList)
+    }
+}
+
+@Composable
+private fun LoginInWithSmileButton(navigateToMovieList: () -> Unit) {
+    Button(
+        onClick = { navigateToMovieList() },
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(bottom = 10.dp)
+    ) {
+        Text(stringResource(id = R.string.login_btn))
+    }
+}
+
+@Composable
+private fun PasswordField() {
+    val passwordState = remember { mutableStateOf(TextFieldValue()) }
+    var passwordVisible by rememberSaveable { mutableStateOf(false) }
+
+    TextField(
+        value = passwordState.value,
+        onValueChange = { passwordState.value = it },
+        modifier = Modifier.padding(bottom = 10.dp),
+        isError = false,
+        placeholder = { Text("Password") },
+        label = {
+            Text(text = stringResource(id = R.string.login_password_text_input_field))
+        },
+
+        visualTransformation = if (passwordVisible) VisualTransformation.None else PasswordVisualTransformation(),
+        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
+        trailingIcon = {
+            val image = if (passwordVisible)
+                Icons.Filled.Visibility
+            else Icons.Filled.VisibilityOff
+
+            // Please provide localized description for accessibility services
+            val description = if (passwordVisible) "Hide password" else "Show password"
```

Comment:
Put string to `strings.xml`

Reply:
I believe this is good on this shape because if I do put it in string.xml , return value of description is wrong ( it says it returns unit instead of String).  And is just a description 
![Screenshot 2022-05-20 213316](https://user-images.githubusercontent.com/36711021/169599163-d1ff94c2-db9d-4f91-a293-d0e82b7ecac3.jpg)

Reply:
![Screenshot 2022-05-20 213701](https://user-images.githubusercontent.com/36711021/169599583-f8720bd7-8c0c-4d66-ac4e-b83936754509.jpg)

Reply:
How have you defined the description? You need to use `stringResource` together with `if else`. f.e. like this `val description = stringResource(id = if (passwordVisible) R.string.hidePassword else R.string.showPassword)`